airflow/docker.go

```diff
@@ -589,6 +590,32 @@ func Deploy(path, name, wsId string, prompt bool) error {
 	// Build our image
 	fmt.Println(messages.COMPOSE_IMAGE_BUILDING_PROMT)
 
+	// parse dockerfile
+	cmds, err := docker.ParseFile(filepath.Join(path, "Dockerfile"))
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse dockerfile: %s", filepath.Join(path, "Dockerfile"))
+	}
+
+	image, tag := docker.GetImageTagFromParsedFile(cmds)
+	if image != messages.VALID_DOCKERFILE_BASE_IMAGE {
+		fmt.Printf("WARNING!!! You are using invalid image name '%s' in your Dockerfile, please replace to %s\n", image, messages.VALID_DOCKERFILE_BASE_IMAGE)
+	}
+
+	// Get valid image tags for platform using Deployment Info request
+	diReq := houston.Request{
+		Query: houston.DeploymentInfoRequest,
+	}
+
+	diResp, err := diReq.Do()
+	if err != nil {
+		return err
+	}
+
+	if !diResp.Data.DeploymentConfig.IsValidTag(tag) {
+		validTags := strings.Join(diResp.Data.DeploymentConfig.GetValidTags(), ",")
+		fmt.Printf("WARNING!!! You are going to push an image using the '%s' tag. This is not recommended. Please base your image off to one of valid image tags: %s\n", tag, validTags)
```

Comment:
If it's not recommended we should ask for confirmation?

Reply:
@schnie what do you think?

Reply:
Yea, seems like they should have to manually agree to move forward.

Reply:
```
➜  t_123 astro-cli deploy sidereal-vacuum-8961
Deploying: sidereal-vacuum-8961
sidereal-vacuum-8961/airflow
Building image...
WARNING!!! You are using invalid image name 'ubuntu' in your Dockerfile, please replace to astronomerinc/ap-airflow
WARNING!!! You are going to push an image using the 'latest' tag. This is not recommended. Please base your image off to one of valid image tags: 1.10.5-alpine3.10-onbuild. Are you sure you want to continue?
 (y/n) n
Cancelling deploy...
```
and
```
➜  t_123 astro-cli deploy sidereal-vacuum-8961
Deploying: sidereal-vacuum-8961
sidereal-vacuum-8961/airflow
Building image...
WARNING!!! You are using invalid image name 'ubuntu' in your Dockerfile, please replace to astronomerinc/ap-airflow
WARNING!!! You are going to push an image using the 'latest' tag. This is not recommended. Please base your image off to one of valid image tags: 1.10.5-alpine3.10-onbuild. Are you sure you want to continue?
 (y/n) y
Sending build context to Docker daemon  10.75kB
Step 1/1 : FROM ubuntu:latest
 ---> 7698f282e524
Successfully built 7698f282e524
Successfully tagged sidereal-vacuum-8961/airflow:latest
Pushing image to Astronomer registry
The push refers to repository [registry.localhost/sidereal-vacuum-8961/airflow]
```
ok?